From d0516fe77c61aea1bfdfb73aaf46413d7cc2685c Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 7 Jul 2017 17:40:51 +0800
Subject: [PATCH 2086/5242] MLK-15925-2 gpu: imx: dpu: fetchdecode: Add a
 helper to report if fd is enabled

commit  9b0d0c0779861204a7dcd4a8c67f1e382789405d from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds a helper so that users may know if the fetchdecode is enabled
or not.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-fetchdecode.c |   12 ++++++++++++
 include/video/dpu.h                   |    1 +
 2 files changed, 13 insertions(+)

diff --git a/drivers/gpu/imx/dpu/dpu-fetchdecode.c b/drivers/gpu/imx/dpu/dpu-fetchdecode.c
index 69f03fa..dc24934 100644
--- a/drivers/gpu/imx/dpu/dpu-fetchdecode.c
+++ b/drivers/gpu/imx/dpu/dpu-fetchdecode.c
@@ -268,6 +268,18 @@ void fetchdecode_layerproperty(struct dpu_fetchdecode *fd, bool enable)
 }
 EXPORT_SYMBOL_GPL(fetchdecode_layerproperty);
 
+bool fetchdecode_is_enabled(struct dpu_fetchdecode *fd)
+{
+	u32 val;
+
+	mutex_lock(&fd->mutex);
+	val = dpu_fd_read(fd, LAYERPROPERTY0);
+	mutex_unlock(&fd->mutex);
+
+	return !!(val & SOURCEBUFFERENABLE);
+}
+EXPORT_SYMBOL_GPL(fetchdecode_is_enabled);
+
 void fetchdecode_clipdimensions(struct dpu_fetchdecode *fd, unsigned int w,
 				unsigned int h)
 {
diff --git a/include/video/dpu.h b/include/video/dpu.h
index 32c77ea..e9f206e 100644
--- a/include/video/dpu.h
+++ b/include/video/dpu.h
@@ -419,6 +419,7 @@ void fetchdecode_clipoffset(struct dpu_fetchdecode *fd, unsigned int x,
 void fetchdecode_clipdimensions(struct dpu_fetchdecode *fd, unsigned int w,
 				unsigned int h);
 void fetchdecode_layerproperty(struct dpu_fetchdecode *fd, bool enable);
+bool fetchdecode_is_enabled(struct dpu_fetchdecode *fd);
 void fetchdecode_framedimensions(struct dpu_fetchdecode *fd, unsigned int w,
 				 unsigned int h);
 void fetchdecode_rgb_constantcolor(struct dpu_fetchdecode *fd,
-- 
1.7.9.5

